frontend/src/components/common/form/StepForm/context.tsx

```diff
@@ -116,7 +119,10 @@ export const StepperFormProvider: React.FC<IStepperFormProviderProps> = ({
   };
 
   const gotoStep = (index: number) => {
-    if (index > 0 && index < steps.length) {
+    if (!validateCurrentStep()) {
+      return false;
+    }
+    if (index >= 0 && index < steps.length) {
```

Comment:
the index can be zero in this case.

Reply:
why can it be zero?

Reply:
the steps are 0 based, and the user is allowed to go back to the first step. This does not apply to the goToNextStep function above.